app/routes/carpool.new.tsx

```diff
@@ -12,109 +22,159 @@ interface Car {
   owner: string;
 }
 
+export async function loader({ request }: LoaderFunctionArgs) {
+  return (await getUser(request)) ?? null;
+}
+
 export async function action({ request }: ActionFunctionArgs) {
   const userId = await requireUserId(request);
   const formData = await request.formData();
 
-  if (formData.get("formType") === "form1") {
-    const name = formData.get("carpoolName")?.toString();
+  const name = formData.get("carpoolName")?.toString();
+  const cars = JSON.parse(formData.get("cars")?.toString() || "[]");
```

Comment:
if you want to properly validate this, you could use zod:
```ts
 const formPayload = Object.fromEntries(await request.formData())
  const schema = z.object({
    carpoolName: z.string(),
    ..otherstuff
  })
  try {
    const data = schema.parse(formPayload)
  } catch (error) {
    // handle me
  }
```
https://egghead.io/blog/validating-remix-form-data-using-zod-and-typescript-in-action-functions